src/coreclr/src/tools/Common/TypeSystem/Common/MethodDesc.cs

```diff
@@ -114,7 +114,7 @@ public int Length
             }
         }
 
-        public bool Equals(MethodSignature otherSignature)
+        public bool Equals(MethodSignature otherSignature, bool allowCovariantReturn = false)
```

Comment:
Agree with Michal.  Equality is a symmetric relation (`x.Equals (y)` and `y.Equals(x)` should both return the same value), adding a boolean flag that breaks symmetry is surprising.

In type theory, the relation between two types (in this case two function signatures) where one is more general than the other (for example because the return type of one function is a supertype of the return type of the other) is called "subsumption".  That might be a good name to use here.

Reply:
Sounds reasonable yes.